Makefile.PL

```diff
@@ -150,7 +151,18 @@ assert_lib(
   LIBS => $opt->{libs},
 );
 
-print "Looks good.\n\n";
+if ($opt->{libs} =~ /mariadb/) {
+  print <<"MSG";
+
+The chosen MySQL client library appears to be MariaDB's. Compilation may fail.
```

Comment:
```suggestion
The chosen MySQL client library appears to be MariaDB's. DBD::mysql doesn't support building against a MariaDB client library.
```

I like this method as it doesn't rely on versions to detect MariaDB (>= 10.0 ) or MySQL (< 10.0) which can be very useful if in the future Oracle would release a MySQL 10.x  (Not unlikely, see the diagram on [this blog post](https://dev.mysql.com/blog-archive/introducing-mysql-innovation-and-long-term-support-lts-versions/) )

Reply:
Not sure what the best text/info is here.

- DBD::mysql requires an Oracle MySQL client library. As the MariaDB client library presents itself as MySQL client library this can be confusing to users. This _might_ get better in the future as `mysql_config` is in `libmariadb-dev-compat` and is optional (at least on Ubuntu 20.04).
- DBD::mysql build with a Oracle MySQL client library can be used to connect to MariaDB.
- Alternatives are DBD::MariaDB and DBD::mysql v4.x. I think only listing the first one (as you did) is probably best, but I'm not sure.
- The MySQL client library must be 8.0 or newer
- Maybe return info on how it was determined to be MariaDB (`mysql_config --libs` matched `mariadb`) to make it easier for people to troubleshoot.

Reply:
I've kept the text descriptive of the behaviour.

Suggesting Oracle's library seems like good advice. Recommending sticking with 4.x might age badly.

It's not unreasonable for someone to _want_ to use this module with MariaDB. DBD::MariaDB has Unicode fixes that might break your application.